msteams-platform/concepts/authentication/nested-authentication.md

```diff
@@ -0,0 +1,190 @@
+
+* You only need to use the MSAL.js library and don’t need the getAuthToken function in TeamsJS.
+* You can call services such as Microsoft Graph with an access token from your client code as an SPA. There’s no need for a middle-tier server.
+* You can use incremental and dynamic consent for scopes.
+* You don't need to preauthorize your hosts (For example, Teams, Office) to call your endpoints.
+
+The NAA model supports a primary identity that includes multiple app identities. Microsoft utilizes this framework in Office Add-Ins and Teams Tabs and Personal apps.
+
+The following table outlines the difference between Teams Microsoft Entra ID SSO and Nested App Authentication:
+
+| Steps | Traditional Teams Microsoft Entra ID SSO | Nested App Authentication |
+| --- | --- | --- |
+| Expose redirect URI | ✔️ | ✔️ * SPA redirect URI necessary |
+| Register API in Microsoft Entra ID | ✔️ |  |
+| Define a custom scope in Microsoft Entra ID | ✔️ |  |
+| Authorize Teams client apps | ✔️ |  |
+| Revise Teams manifest | ✔️ | Recommended helping IT admins in providing consent through the Admin Portal |
+| Acquire access token through Teams JS SDK | ✔️ |  |
+| Acquire access token through MSAL.js |  | ✔️ * Requires new msalconfig property: supportsNestedAppAuth |
+| Solicit user consent for more permissions | ✔️ |  |
+| Conduct an OBO exchange on the server | ✔️ |  |
+
+## Scenarios
+
+| Scenario | description |
+| --- | --- |
+| **Consenting to SSO (and other permissions)** | Tom, a new member of the Contoso design team, needs to use Mural in Teams to collaborate on whiteboards. Upon first use, a dialog prompts Tom to grant permissions, including reading their profile for their avatar (User.Read). After tom consents, Tom can use Mural seamlessly in future meetings across devices. |
+| **Reauthentication or Conditional Access step-up auth** |  Tom, while working from Australia, encounters a conditional access trigger requiring multifactor authentication (MFA) to access Mural in Teams. A dialog informs Tom of the more verification needed, leading them through the MFA process to continue using Mural. |
+| **Errors** | Tom faces an error with Mural displaying an 'Oh no!' page due to an issue retrieving account information. A retry button prompts Tom to reauthenticate, only to find that the system administrator blocked access to Mural. |
+
+## Configure nested authentication
+
+> [!NOTE]
+> Nested authentication is in developer preview and not supported by all host environments. Developers need to check the support status using the Teams JS SDK and provide a fallback experience for unsupported environments.
```

Comment:
Some more prescriptive guidance we can give:

developers should call this API: (https://github.com/OfficeDev/microsoft-teams-library-js/blob/386faf29c6ab698ea2c2a831cfcdb2fa7acc8d2f/apps/teams-test-app/src/components/NestedAppAuthAPIs.tsx#L8)
to extract the value, when “true” they should call MSAL for the NAA flow, and when “false” they should use their existing token fetching call."